lib/rate-limiter.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const Bluebird = require('bluebird');
+
+exports.limit = (request, client, defaultRate) => {
```

Comment:
Yeah I think it could all be together. That way, you'll have more integration-like tests to have more confidence that it works end-to-end, and you won't have to stub things out like the `request` object.